src/bootstrap/src/core/build_steps/setup.rs

```diff
@@ -619,10 +626,110 @@ fn create_vscode_settings_maybe(config: &Config) -> io::Result<bool> {
             }
             _ => "Created",
         };
-        fs::write(&vscode_settings, RUST_ANALYZER_SETTINGS)?;
+        fs::write(&vscode_settings, VSCODE_SETTINGS)?;
         println!("{verb} `.vscode/settings.json`");
     } else {
-        println!("\n{RUST_ANALYZER_SETTINGS}");
+        println!("\n{VSCODE_SETTINGS}");
+    }
+    Ok(should_create)
+}
+
+/// Sets up or displays `src/etc/zed_settings.json`
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct Zed;
+
+impl Step for Zed {
+    type Output = ();
+    const DEFAULT: bool = true;
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("zed")
+    }
+    fn make_run(run: RunConfig<'_>) {
+        if run.builder.config.dry_run() {
+            return;
+        }
+        if let [cmd] = &run.paths[..] {
+            if cmd.assert_single_path().path.as_path().as_os_str() == "zed" {
+                run.builder.ensure(Zed);
+            }
+        }
+    }
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        let config = &builder.config;
+        if config.dry_run() {
+            return;
+        }
+        while !t!(create_zed_settings_maybe(config)) {}
+    }
+}
```

Comment:
nit: This makes it more readable

```suggestion
impl Step for Zed {
    type Output = ();
    const DEFAULT: bool = true;

    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
        run.alias("zed")
    }

    fn make_run(run: RunConfig<'_>) {
        if run.builder.config.dry_run() {
            return;
        }
        if let [cmd] = &run.paths[..] {
            if cmd.assert_single_path().path.as_path().as_os_str() == "zed" {
                run.builder.ensure(Zed);
            }
        }
    }

    fn run(self, builder: &Builder<'_>) -> Self::Output {
        let config = &builder.config;
        if config.dry_run() {
            return;
        }
        while !t!(create_zed_settings_maybe(config)) {}
    }
}
```

Reply:
@onur-ozkan It appears that the spacing has been fixed.